parser/src/lexer/mod.rs

```diff
@@ -78,17 +78,25 @@ pub enum Token {
     // --------------------------------------------------------------------------------------------
     /// Identifiers should start with alphabet followed by one or more alpha numeric characters
     /// or an underscore.
-    Ident(String),
+    Ident(Symbol),
     /// A reference to an identifier used for a section declaration, such as the random values
     /// array or a trace segment like "main" or "aux".
-    DeclIdentRef(String),
+    DeclIdentRef(Symbol),
+    /// A function identifier
+    FunctionIdent(Symbol),
     /// Integers should only contain numeric characters.
     Num(u64),
 
     // DECLARATION KEYWORDS
     // --------------------------------------------------------------------------------------------
-    /// Used to declare AIR constraints module.
+    /// Used to declare AIR program.
     Def,
+    /// Used to declare AIR module.
+    Module,
```

Comment:
In this particular instance I'm cool with either form, but I believe in our previous discussions, the topic of aggressive shorthand for things came up, and it sounded like the general consensus was to prefer more descriptive keywords unless there is a benefit to the shorthand. For example, `enf` is meaningless as a keyword (i.e. unless you already know that it is shorthand for "enforce"), but if you are writing a bunch of constraints, the shorthand is convenient, and is less noisy than the non-shorthand form, so it balances out. However, for less frequently used keywords, the shorthand is just unnecessarily concise (`ev` for example isn't even the typical shorthand for evaluator/evaluation, which would be "eval", so there is some cognitive overhead when encountered). For others, the shorthand is so common in the industry that there is no confusion about what they generally indicate (e.g. `def`, `fn`). In this instance I erred on the side of more descriptive, but `mod` is common enough in the industry that I don't have a particular problem with it vs `module`, so if we prefer `mod` here, that's an easy change to make. 

While I'm on the topic, `def` is a bit of an odd choice for module declarations, since it is typically used for function declarations in other languages where `def` or `define` is a keyword (e.g. Python, Elixir, Scheme/LISP, BASIC), but I think there is benefit in distinguishing between the root module and a library module, and `def` is already part of the language syntax, so changing it now is probably not worth it. I think I'd probably use `program`/`prog` for the root module, since it constitutes an AirScript "program", but ultimately it doesn't matter that much. I'm a programming language design/implementation nerd, so I probably care more about these things than is strictly necessary.

Reply:
My person preference (and not a strong one) is to go with `def/mod` - or if we don't like `def`, another option could be just `air` (I'm not sure about `prog` since this is not really a program - more like a set of polynomial equations).

Reply:
FYI: I've gone ahead and just updated everything to use `mod` for now